tests/unit/parsers/test.locale-messagesjson.js

```diff
@@ -266,6 +266,27 @@ describe('LocaleMessagesJSONParser', () => {
     );
   });
 
+  it('should not contain the words "mozilla" or "firefox" on localized extension name', () => {
+    const addonLinter = new Linter({ _: ['bar'] });
+    const localeMessagesJSONParser = new LocaleMessagesJSONParser(
+      `{
+  "extensionName": {
+    "message": "My Awesome Ext for Mozilla Firefox",
+    "description": "Name of the extension."    
+  }
+}`,
+      addonLinter.collector,
+      { name: '__MSG_extensionName__' }
+    );
+
+    localeMessagesJSONParser.parse();
+    expect(localeMessagesJSONParser.isValid).toEqual(false);
+    const { warnings } = addonLinter.collector;
+    expect(warnings[0].code).toEqual(
+      messages.PROP_NAME_MUST_NOT_CONTAIN_MOZILLA_OR_FIREFOX.code
+    );
+  });
```

Comment:
Could you add another check here that makes sure the matched filename is `messages.json`?

Reply:
Done!